src/escalus_connection.erl

```diff
@@ -239,7 +254,31 @@ get_stanza_safe(Client, Timeout, Pred) ->
                               continue
                       end
               end,
-    receive_stanzas(Client, Timeout, Handler).
+    Safe = maps:get(safe, Options, false),
+    WithMetadata = maps:get(with_metadata, Options, false),
+    Timeout = maps:get(timeout, Options, ?TIMEOUT),
+    case receive_stanzas(Client, Timeout, Handler) of
+        {error, timeout} when Safe ->
```

Comment:
I've not read the context thoroughly, but this seems a bit counterintuitive - we throw when `Safe`, but return by value when not `Safe`?

Reply:
You are right, it should be the opposite, it is still not finished and definitely needs more testing :-)